src/sql-parser/src/parser.rs

```diff
@@ -1915,6 +1927,67 @@ impl<'a> Parser<'a> {
         }))
     }
 
+    fn parse_kafka_security(&mut self) -> Result<KafkaSecurityOptions, ParserError> {
+        Ok(match self.expect_one_of_keywords(&[SSL, SASL, SASLSSL])? {
+            Keyword::Ssl => {
+                let (mut key, mut certificate, mut passphrase) = (None, None, None);
+                while let Ok(keyword) = self.expect_one_of_keywords(&[KEY, CERTIFICATE, PASSWORD]) {
+                    match keyword {
+                        Keyword::Key => {
+                            let _ = self.consume_token(&Token::Eq);
+                            key = Some(self.parse_object_name()?);
+                        }
+                        Keyword::Certificate => {
+                            let _ = self.consume_token(&Token::Eq);
+                            certificate = Some(self.parse_object_name()?);
+                        }
+                        Keyword::Password => {
+                            let _ = self.consume_token(&Token::Eq);
+                            passphrase = Some(self.parse_object_name()?);
+                        }
+                        _ => unreachable!(),
+                    }
+                }
+                KafkaSecurityOptions::SSL {
+                    key,
+                    certificate,
+                    passphrase,
+                }
+            }
+            Keyword::Sasl => self.parse_kafka_sasl(false)?,
+            Keyword::SaslSsl => self.parse_kafka_sasl(true)?,
+            _ => unreachable!(),
+        })
+    }
+
+    fn parse_kafka_sasl(&mut self, ssl: bool) -> Result<KafkaSecurityOptions, ParserError> {
+        self.expect_keyword(MECHANISM)?;
+        let mechanism = match self.expect_one_of_keywords(&[PLAIN, SCRAMSHA256, SCRAMSHA512])? {
```

Comment:
nit: if this variable only accepts `&'static str`, that should be its type, and the type of the field in the struct that receives it. No need to do a song and dance making this more flexible than it needs to be.

Reply:
`walkabout` would require more extending for this field to go from `String` to `&str` with or without a lifetime annotation. Not sure if that is the better change than having some `.to_string()` here.

Reply:
Ah, I didn't realize `walkabout` had that limitation; no sweat